main.go

```diff
@@ -181,6 +181,10 @@ func wrapMain() error {
 
 	secrets := map[string]interface{}{}
 	for k, v := range vargs.Secrets {
+		if v == nil {
```

Comment:
This would only check for `null` values in the yaml. You probably want to do a type check for string `str, ok := v.(string)` and then check if `str == ""`